windows-containers/productization.md

```diff
@@ -0,0 +1,188 @@
+---
+title: windows-containers-productization
+authors:
+  - "@ravisantoshgudimetla"
+  - "@aravindhp"
+reviewers:
+  - "@crawford"
+  - "@sdodson"
+approvers:
+  - "@crawford"
+  - "@sdodson"
+creation-date: 2019-08-30
+last-updated: 2019-09-03
+status: implementable
+---
+
+# Windows Containers Productization
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+The intent of this enhancement is to allow a cluster administrator to add a
+Windows worker node with a prescribed configuration to an OpenShift cluster as a
+day 2 operation and enable scheduling of Windows workloads.
+
+## Motivation
+
+The main motivation behind this enhancement is to satisfy customer
+requirement of being able to run Windows workloads on OpenShift clusters.
+
+### Goals
+
+As part of this enhancement we plan to do the following:
+* Prepare an already provisioned Windows worker node to join the cluster
+* Perform all the required steps within the node for it to be a OpenShift worker
+node
+* Administrator initiated upgrade of all OpenShift related components
+(kubelet, OVN and Windows Machine Config Bootstrapper) in the node
+
+### Non-Goals
+
+As part of this enhancement we do not plan to support:
+* Windows node provisioning or de-provisioning
+* Installing the container runtime in the Windows node
+* Windows operating system upgrades
+* Node management like reboots and node draining
+* OpenShift Builds
+* Network configuration
+  * The details of this will be worked out as part of the enhancement
+    proposal for "OVN Plumbing for Hybrid Linux+Windows Clusters GA"
+
+
+## Proposal
+
+The two main pieces that will allow us to achieve our goals is a Windows Scale
+Up (WSU) Ansible playbook and a Windows Machine Config Bootstrapper (WMCB)
+binary. The Ansible playbook will collect information from the cluster and
+transfer it to the Windows node. The WMCB will then perform the necessary
+steps on the node for it to join the cluster.
+
+### Justification
+
+The reason for having a Ansible playbook and on-node executable split is to have
```

Comment:
Perhaps we can re-word this to `consistent user experience`

Reply:
Done